api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java

```diff
@@ -150,4 +152,18 @@
      *             if the given value was {@code null}
      */
     T convert(String value) throws IllegalArgumentException, NullPointerException;
+
+    /**
+     * Convert the given config node (may be an object, list or value) to the specified type.
```

Comment:
If this config node is a node instead of leaf, I assume an `IllegalArugumentException` should be thrown.

Reply:
A converter may be capable of converting object nodes or list nodes as well. If the converter converts to a POJO for example, accepting an Object node is absolutely fine